docs/Contributing-back.rst

```diff
@@ -0,0 +1,496 @@
+since **only contributions that fulfill all of the requirements discussed below
+are eligible for being merged in**:
+
+* unit tests must still pass (see :ref:`contributing_back_review_process_travis`)
+
+  * more tests must be added when appropriate (see :ref:`contributing_back_review_process_adding_tests`)
+
+* backward compatibility should be retained (see :ref:`contributing_back_review_process_backward_compatibility`)
+* code style must be kept consistent (see :ref:`contributing_back_review_process_code_style`)
+
+  * easyconfigs should be kept consistent across versions & toolchains (see :ref:`contributing_back_review_process_review_pr`)
+
+* test reports must be submitted for easyconfig contributions (see :ref:`contributing_back_review_process_test_reports`)
+
+
+.. _contributing_back_review_process_travis:
+
+Test suite (Travis)
+~~~~~~~~~~~~~~~~~~~
+
+Each pull request is tested automatically by Travis, and the test result
+is reported in the pull request.
+
+**Only pull requests that have been tested and approved by Travis are
+eligible for being merged!**
+
+Note that Travis will only run the *unit test suite* for that particular repository.
+That is, for easyconfig contributions it does *not* include
+actually building and installing software.
+
+.. _contributing_back_review_process_adding_tests:
+
+Adding tests
+++++++++++++
+
+An implicit requirement for contributions, in particular contributions to the EasyBuild framework,
+is that they **must be accompanied by additional tests or test cases**.
+
+For new features or enhancements, a dedicated test (case) must be added
+which verifies that the feature implementation works as expected.
+
+For bug fixes, a test (case) must be added that triggers the code path where the bug manifested,
+and which verifies that the bug was indeed fixed.
+
+Tests not only confirm that the implementation is correct, it also helps to ensure
+that any future changes will not affect semantics.
+
+
+.. _contributing_back_review_process_backward_compatibility:
+
+Backward compatibility
+~~~~~~~~~~~~~~~~~~~~~~
+
+**Contributions should retain backward compatibility**, i.e., they should *not*
+make any changes that alter the (default) semantics of the existing code base.
+Of course, enhancements to existing code that retain backward compatibility can be made.
+
+One exception to this rule is *bug fixes*, where the whole point is usually
+to fix functionality that was implemented incorrectly.
+
+This also applies to existing easyconfig files; for example, the versions
+of dependencies should *not* be altered. Adding dependencies that where missing
+or otherwise enhancing existing easyconfigs (e.g., adding extra extensions,
+enabling additional features, etc.) are usually considered acceptable.
+
+In any case, changes that break backward compatibility have to be motivated
+well with technical arguments, and must be approved by the EasyBuild maintainers.
```

Comment:
"in any case" should be removed anyway. It is too informal considering the previous context (pretty much a voice change).